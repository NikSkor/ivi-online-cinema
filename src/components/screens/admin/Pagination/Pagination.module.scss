@import '../../../../styles/vars.scss';

.paginationBlock {
  display: flex;
  gap: 35px;
  align-items: center;
  margin-top: 30px;
  margin-bottom: 20px;
  color: #fff;
}

.list {
  display: flex;
  font-weight: 500;
  font-size: 14px;
  line-height: 130%;
  // color: #3F4E65;
  align-items: center;
  gap: 9px;
  // height: 14px;
}

.itemActive {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-left: 14px;
  padding-right: 14px;
  gap: 5px;
  color: $red;
  font-size: 14px;
  line-height: 130%;
  // background-color: linear-gradient(90deg, rgba(255, 198, 80, 0.3) 0%, rgba(254, 202, 110, 0.3) 97.25%);
  // border-radius: 100%;
}