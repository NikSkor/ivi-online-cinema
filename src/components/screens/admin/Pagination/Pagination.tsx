import React, { FC } from 'react';
import style from './Pagination.module.scss';
import ArrowLeft from '../../../../../public/arrow-left.svg';
import ArrowRight from '../../../../../public/arrow-right.svg';
import { useAppDispatch} from '@/store/hooks';
import { newPage } from '@/store/slices/adminSlice';
import Image from 'next/image';



interface IPageNumbers {
  pagesSum: number,
  pageActive: number,
  getPage: Function
}


const Pagination: FC<IPageNumbers> = ({pagesSum, pageActive, getPage}) => {
  
  const dispatch = useAppDispatch();

  let arrCreate = (pages: number) => {
  let res: number[] = [1];

  if (pages === 0) res.length = 0;
  
  for (let i = 2; i <= pages; i++) {
    res.push(i);
  }

  return res;
  }

  const pageNumLeftHandler = (e : React.MouseEvent) => {
    e.preventDefault();
    if (pageActive - 1 === 0) {
      dispatch(newPage(1));
    } else {
      dispatch(newPage(pageActive-1));
    }
    getPage();
  }

  const pageNumRightHandler = (e : React.MouseEvent) => {
    e.preventDefault();
    if (pageActive + 1 >= pagesSum) {
      dispatch(newPage(pagesSum));
    } else {
      dispatch(newPage(pageActive+1));
    }
    getPage();
  }

  const pageNumberClickHandler = (e: any)=> {
    dispatch(newPage(+e.target.textContent));
    getPage();
  }

  let pagesArr: number[] = arrCreate(pagesSum);

  if (pagesSum === 0 || pagesSum === 1) return null;


  return (
    <div className={style.paginationBlock}>
      <button className={style.arrowLeft} onClick={pageNumLeftHandler}>
        <Image src={ArrowLeft} alt="Стрелка влево" />
      </button>
      <ul className={style.list}>
        {
          pagesArr.map((item) => {
            return (item === pageActive) ? <li key={item} className={`${style.item} ${style.itemActive}`}
              onClick={pageNumberClickHandler}
            >{item}</li> 
            : <li key={item} className={style.item}
              onClick={pageNumberClickHandler}
            >{item}</li>
          })
        }
      </ul>
      <button onClick={pageNumRightHandler}>
        <Image src={ArrowRight} alt="Стрелка вправо" />
      </button>
    </div>
    
  )
}

export default Pagination;