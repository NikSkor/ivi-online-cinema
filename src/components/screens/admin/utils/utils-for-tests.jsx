import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import adminReducer from '../../../../store/slices/adminSlice';


export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { admin: adminReducer},
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}