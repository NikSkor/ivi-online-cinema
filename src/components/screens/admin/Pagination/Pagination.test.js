import { cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Pagination from '@/components/screens/admin/Pagination/Pagination';
import { renderWithProviders } from '../utils/utils-for-tests';

jest.mock('next/router', () => require('next-router-mock'));

describe('test Pagination component', () => {
  afterEach(cleanup);

  test('there are the arrow btns', () => {
    renderWithProviders(
      <Pagination pagesSum={5} pageActive={2} getPage={() => {}} />
    );
    const btns = screen.getAllByRole('button');
    btns.forEach((item)=> {
      expect(item).toBeInTheDocument();
      expect(item).toMatchSnapshot();
    })
  });

  test('component renders props correctly, truthy values', () => {
    const tree = renderWithProviders(
      <Pagination pagesSum={5} pageActive={2} getPage={()=>{}} />
    );
    expect(tree).toMatchSnapshot();
  });
  test('component renders props correctly, falsy values', () => {
    const tree = renderWithProviders(<Pagination />);
    expect(tree).toMatchSnapshot();
  });
});
