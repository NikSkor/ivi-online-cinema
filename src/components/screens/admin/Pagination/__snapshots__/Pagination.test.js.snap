// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Pagination component component renders props correctly, falsy values 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="paginationBlock"
      >
        <button
          class="arrowLeft"
        >
          <img
            alt="Стрелка влево"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style="color: transparent;"
            width="24"
          />
        </button>
        <ul
          class="list"
        >
          <li
            class="item"
          >
            1
          </li>
        </ul>
        <button>
          <img
            alt="Стрелка вправо"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style="color: transparent;"
            width="24"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="paginationBlock"
    >
      <button
        class="arrowLeft"
      >
        <img
          alt="Стрелка влево"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="24"
        />
      </button>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          1
        </li>
      </ul>
      <button>
        <img
          alt="Стрелка вправо"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="24"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`test Pagination component component renders props correctly, truthy values 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="paginationBlock"
      >
        <button
          class="arrowLeft"
        >
          <img
            alt="Стрелка влево"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style="color: transparent;"
            width="24"
          />
        </button>
        <ul
          class="list"
        >
          <li
            class="item"
          >
            1
          </li>
          <li
            class="item itemActive"
          >
            2
          </li>
          <li
            class="item"
          >
            3
          </li>
          <li
            class="item"
          >
            4
          </li>
          <li
            class="item"
          >
            5
          </li>
        </ul>
        <button>
          <img
            alt="Стрелка вправо"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style="color: transparent;"
            width="24"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="paginationBlock"
    >
      <button
        class="arrowLeft"
      >
        <img
          alt="Стрелка влево"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="24"
        />
      </button>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          1
        </li>
        <li
          class="item itemActive"
        >
          2
        </li>
        <li
          class="item"
        >
          3
        </li>
        <li
          class="item"
        >
          4
        </li>
        <li
          class="item"
        >
          5
        </li>
      </ul>
      <button>
        <img
          alt="Стрелка вправо"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="24"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`test Pagination component there are the arrow btns 1`] = `
<button
  class="arrowLeft"
>
  <img
    alt="Стрелка влево"
    data-nimg="1"
    decoding="async"
    height="24"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
    style="color: transparent;"
    width="24"
  />
</button>
`;

exports[`test Pagination component there are the arrow btns 2`] = `
<button>
  <img
    alt="Стрелка вправо"
    data-nimg="1"
    decoding="async"
    height="24"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
    style="color: transparent;"
    width="24"
  />
</button>
`;
